src/net_processing.cpp

```diff
@@ -2288,6 +2290,8 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 if (nDoS > 0) {
                     LOCK(cs_main);
                     Misbehaving(pfrom->GetId(), nDoS);
+                } else {
+                    pfrom->fDisconnect = true;
```

Comment:
Same here.

Reply:
Removed the `else` statement so it will always disconnect on an invalid block header.